readdy/main/model/Actions.cpp

```diff
@@ -47,6 +48,447 @@ reactions::UncontrolledApproximation::UncontrolledApproximation(scalar timeStep)
+    {
+        const auto result = util::integration::integrateAdaptive(rhsIntegrand, 0., rhsInteractionRadius, desiredRelativeError, maxiter);
+        const auto achievedRelativeError = result.second / result.first;
+        if (achievedRelativeError> desiredRelativeError) {
+            readdy::log::warn("Integration of potentials yielded larger error ({}) than desired ({}).",
+                              achievedRelativeError, desiredRelativeError);
+        }
+        effectiveRhsInteractionVolume = result.first;
+    }
+
+    // effective lhs reaction volume
+    {
+        const auto result = util::integration::integrateAdaptive(lhsIntegrand, 0., reactionRadius, desiredRelativeError, maxiter);
+        const auto achievedRelativeError = result.second / result.first;
+        if (achievedRelativeError> desiredRelativeError) {
+            readdy::log::warn("Integration of potentials yielded larger error ({}) than desired ({}).",
+                              achievedRelativeError, desiredRelativeError);
+        }
+        effectiveLhsReactionVolume = result.first;
+    }
+
+    // effective rhs reaction volume
+    {
+        const auto result = util::integration::integrateAdaptive(rhsIntegrand, 0., reactionRadius, desiredRelativeError, maxiter);
+        const auto achievedRelativeError = result.second / result.first;
+        if (achievedRelativeError> desiredRelativeError) {
+            readdy::log::warn("Integration of potentials yielded larger error ({}) than desired ({}).",
+                              achievedRelativeError, desiredRelativeError);
+        }
+        effectiveRhsReactionVolume = result.first;
+    }
+
+    // FusionFission needs a cumulative distribution
+    if (reversibleType == ReversibleType::FusionFission) {
+        const size_t nPoints = 1000;
+        fissionRadii.reserve(nPoints);
+        const scalar dr = reactionRadius / static_cast<scalar>(nPoints);
+        scalar r = 0.;
+        for (size_t i = 0; i < nPoints + 1; ++i) {
+            fissionRadii.push_back(r);
+            r += dr;
+        }
+        for (size_t i = 0; i < fissionRadii.size(); ++i) {
+            const auto result = util::integration::integrateAdaptive(lhsIntegrand, fissionRadii.front(),
+                                                                     fissionRadii[i], desiredRelativeError, maxiter);
+            const auto cumvalue = result.first;
+            const auto achievedRelativeError = result.second /  result.first;
+            if (achievedRelativeError > desiredRelativeError) {
+                readdy::log::warn("Integration of lhs potentials yielded larger error ({}) than desired ({}).",
+                                  achievedRelativeError, desiredRelativeError);
+            }
+            cumulativeFissionProb.push_back(cumvalue);
+        }
+        // normalize fission probability to 1
+        for (auto &p : cumulativeFissionProb) {
+            p = p / cumulativeFissionProb.back();
```

Comment:
can `cumulativeFissionProb.back()` become very small? we should probably have a cutoff

Reply:
Yes, that might happen. E.g. if one has a LJ potential that covers most of the reactive volume (diverging such that e^{-U} is very small). So a cutoff makes sense, in this case all values in this vector should be 0. except for the last, which then should be 1. to yield a proper cumulative.